src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -0,0 +1,630 @@
+        {
+            long unixTimeSeconds = timestamp.ToUnixTimeSeconds();
+            return WriteAsOctal(unixTimeSeconds, destination, location, length);
+        }
+
+        // Writes the specified text as an ASCII string aligned to the left, and returns its checksum.
+        private static int WriteAsAsciiString(string str, Span<byte> buffer, int location, int length)
+        {
+            byte[] bytes = Encoding.ASCII.GetBytes(str);
+            return WriteLeftAlignedBytesAndGetChecksum(bytes.AsSpan(), buffer.Slice(location, length));
+        }
+
+        // Gets the special name for the 'name' field in an extended attribute entry.
+        // Format: "%d/PaxHeaders.%p/%f"
+        // - %d: The directory name of the file, equivalent to the result of the dirname utility on the translated pathname.
+        // - %p: The current process ID.
+        // - %f: The filename of the file, equivalent to the result of the basename utility on the translated pathname.
+        private string GenerateExtendedAttributeName()
+        {
+            string? dirName = Path.GetDirectoryName(_name);
+            dirName = string.IsNullOrEmpty(dirName) ? "." : dirName;
+
+            int processId = Environment.ProcessId;
+
+            string? fileName = Path.GetFileName(_name);
+            fileName = string.IsNullOrEmpty(fileName) ? "." : fileName;
+
+            string trailingSeparator = (_typeFlag is TarEntryType.Directory or TarEntryType.DirectoryList) ?
+                $"{Path.DirectorySeparatorChar}" : string.Empty;
+
+            return string.Format(PaxHeadersFormat, dirName, processId, fileName, trailingSeparator);
+        }
+
+        // Gets the special name for the 'name' field in a global extended attribute entry.
+        // Format: "%d/GlobalHead.%p/%f"
+        // - %d: The path of the $TMPDIR variable, if found. Otherwise, the value is '/tmp'.
+        // - %p: The current process ID.
+        // - %n: The sequence number of the global extended header record of the archive, starting at 1. In our case, since we only generate one, the value is always 1.
+        // If the path of $TMPDIR makes the final string too long to fit in the 'name' field,
+        // then the TMPDIR='/tmp' is used.
+        private static string GenerateGlobalExtendedAttributeName()
+        {
+            string? tmpDir = Environment.GetEnvironmentVariable("TMPDIR");
+            if (string.IsNullOrWhiteSpace(tmpDir))
+            {
+                tmpDir = "/tmp";
+            }
+            else if (Path.EndsInDirectorySeparator(tmpDir))
+            {
+                tmpDir = Path.TrimEndingDirectorySeparator(tmpDir);
+            }
+            int processId = Environment.ProcessId;
+
+            string result = string.Format(GlobalHeadFormat, tmpDir, processId);
+            if (result.Length >= FieldLengths.Name)
+            {
+                result = string.Format(GlobalHeadFormat, "/tmp", processId);
+            }
```

Comment:
```suggestion
            string result = $"{tmpDir}/GlobalHead.{Environment.ProcessId}.1";
            if (result.Length >= FieldLengths.Name)
            {
                result = $"/tmp/GlobalHead.{Environment.ProcessId}.1";
            }
```
and maybe some assert in the if block that tmpDir is so much longer than /tmp that the new string is guaranteed to fit?